### PR TITLE
Geckodriver

### PR DESCRIPTION
**Pull request by aliking**
update everything to selenium 3 and replace that thing where local server starts with rake.

### CODE REVIEW

File: selenium-connect.gemspec [hunk 1]
```diff
@@ -14,10 +14,10 @@ Gem::Specification.new do |s|
...
 
-  s.add_dependency 'selenium-webdriver', '2.53.1'
+  s.add_dependency 'selenium-webdriver'
   s.add_dependency 'rake'
-  s.add_dependency 'sauce'
+  s.add_dependency 'sauce', '~> 3.7'
   s.add_dependency 'sauce-connect'
   s.add_dependency 'sauce_whisk'
-  s.add_dependency 'appium_lib', '~> 6.0.0'
+  s.add_dependency 'appium_lib', '~> 8.0.0'
```
Comment (jaysirju) [inline selenium-connect.gemspec:22]: Have you tried running the test_selenium_ios tests with appium 8?  If not, I would suggest not upgrading this until it's been tested out.

File: lib/selenium_connect.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Encoding: utf-8
 
 require 'selenium_connect/job'
-require 'selenium_connect/server'
+#require 'selenium_connect/server'\
```
Comment (jaysirju) [inline lib/selenium_connect.rb]: Commented out code makes me sad :P

File: lib/selenium_connect/runner.rb [hunk 3]
```diff
@@ -42,6 +42,14 @@ def init_driver
         driver = AppiumLib.new(config).launch
       elsif config.host == 'browserstack'
         driver = BrowserStack.new(config).launch
+      elsif config.browser == 'firefox'
+        config.port = nil if config.browser == 'testdroid'
+        driver = Selenium::WebDriver.for(
+            :remote,
+            url: set_server_url,
+            #marionette: true,
```
Comment (jaysirju) [inline lib/selenium_connect/runner.rb]: remove the commented out capability.  It can be added back later if needed

**Comment by jaysirju** [comment]
A couple items, but looks good otherwise

File: selenium-connect.gemspec (hunk 1 above)
Comment (aliking) [reply selenium-connect.gemspec:22]: That is a good point, they may well not. This was really just some acrobatics to work around appium not being updated to use selenium 3. I just checked and it is now, so I can simplify that. As for the ios tests, I don't want to let them hold us back, so what I'll do is set a permanent tag for v3.7.1 of selenium connect, and not update test selenium ios.

File: lib/selenium_connect.rb (hunk 2 above)
Comment (aliking) [reply lib/selenium_connect.rb]: oh yeah, I also deleted the code this use to point to...

**Pull request merged**